modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -543,6 +550,32 @@ public static Object consumeFirstArgument(Object args0, Object args1) {
         return args1;
     }
 
+    /** Converts timestamp string into a timestamp with local time zone. */
+    public static @Nullable Long toTimestampWithLocalTimeZone(@Nullable String v, TimeZone timeZone) {
+        return SqlFunctions.toTimestampWithLocalTimeZone(v, timeZone);
+    }
+
+    /** Converts timestamp string into a timestamp with local time zone. */
+    public static @Nullable Long toTimestampWithLocalTimeZone(@Nullable String v, String format, TimeZone timeZone) {
+        if (v == null) {
+            return null;
+        }
+
+        Objects.requireNonNull(format, "format");
+        Objects.requireNonNull(timeZone, "timeZone");
+
+        // TODO https://issues.apache.org/jira/browse/IGNITE-25320 reuse to improve performance.
+        DateParseFunction function = new DateParseFunction();
+        long ts = function.parseTimestamp(format, v);
+        Instant instant = Instant.ofEpochMilli(ts);
+
+        // Adjust instant millis
+        ZoneRules rules = timeZone.toZoneId().getRules();
+        ZoneOffset offset = rules.getOffset(instant);
+        Instant adjusted = instant.minus(offset.getTotalSeconds(), ChronoUnit.SECONDS);
+        return adjusted.toEpochMilli();
```

Comment:
May be this code can be replaced with the following? (needs to be checked carefully)
```suggestion
    return IgniteSqlDateTimeUtils.subtractTimeZoneOffset(ts, timeZone);
```

Reply:
Thanks, fixed.